app/models/course.rb

```diff
@@ -117,6 +118,31 @@ def root_folder?
     end
   end
 
+  def initialize_duplicate(duplicator, other)
+    self.start_at += duplicator.time_shift
+    self.end_at += duplicator.time_shift
+    self.title = duplicator.new_course_title
+
+    # This also duplicates assessments.
+    self.lesson_plan_items = duplicator.duplicate(other.lesson_plan_items.map(&:actable)).
+                             map(&:acting_as)
+    self.lesson_plan_milestones = duplicator.duplicate(other.lesson_plan_milestones)
+
+    # Find material_folders without owners and only duplicate those.
+    # This must be done after duplicating assessments.
+    # Do not try duplicating all folders at once. Parent IDs do not seem to be populated by the
+    # edge gem until the database entries are created.
+    material_folders_to_duplicate = other.material_folders.without_owners
+    self.material_folders = duplicator.duplicate(material_folders_to_duplicate).compact
```

Comment:
In future not everything might be selected, so duplicating the unselected source folders will return `nil`.